src/commands/install-deps.yml

```diff
@@ -45,22 +38,35 @@ steps:
           APP_BUNDLER_VERSION=<<parameters.bundler-version>>
         fi
 
-        if ! echo $(bundle version)  | grep -q $APP_BUNDLER_VERSION; then
+        if ! echo $(bundle version) | grep -q $APP_BUNDLER_VERSION; then
           echo "Installing bundler $APP_BUNDLER_VERSION"
           gem install bundler:$APP_BUNDLER_VERSION
         else
           echo "bundler $APP_BUNDLER_VERSION is already installed."
         fi
 
+        test -f Gemfile.lock || bundle lock
+
+  - when:
+      condition: <<parameters.with-cache>>
+      steps:
+        - restore_cache:
+            keys:
+              - << parameters.key >>-{{ .Branch }}-{{ checksum "./Gemfile.lock"  }}
```

Comment:
@elia Thanks for updating! We're still going to run into the issue from #26, though. Running through the possible scenarios, I think it ends up being the same cache hit rate if we switch to `.Environment.CIRCLE_SHA1` instead of `checksum` here. Some scenarios:

1) Create a new PR from a branch:

With both approaches it will end up falling back to the most general key `<< parameters.key >>-`, pulling the most recent bundler cache from any other branch.

2) Updating a branch with no gem changes:

With your approach it will hit on the first, most specific key. With the SHA, it will hit on the second key and create a new key afterwards. Both will have the same vendored gems from the cache, though, so there won't be any work for bundler to do. Storing a new cache takes around 10 seconds, though it only happens for a single during a build. Other workers will return early when they see the cache key already exists.

3) Updating a branch with gem changes:

Both will fall back to the second key, pulling a slightly stale cache and bundler will have to update vendored gems. Both will then push a new cache.

The differences here seem negligible, so why not go the approach that covers more use cases? There's slightly more work for CircleCI needing to store more caches, but maybe @gmemstr can clarify whether that is a consideration on their end.

Reply:
I think in this case it's a relatively small lift for storing these caches, although I'm not sure about the specific impact it might have.